### PR TITLE
cleanup: Reduce cyclomatic complexity of topology_test

### PR DESCRIPTION
**Pull request by Yuggupta27**
Reduce redundant code to make the test simpler.

Updates: #975 #1157 

Signed-off-by: Yug <yuggupta27@gmail.com>

### CODE REVIEW

File: internal/util/topology_test.go [hunk 1]
```diff
@@ -23,7 +23,18 @@ import (
 	"github.com/container-storage-interface/spec/lib/go/csi"
 )
 
-// nolint: gocyclo
+func checkErr(t *testing.T, msg string, err error) {
```
Comment (Madhu-1) [inline internal/util/topology_test.go:26]: make it as checkError

File: internal/util/topology_test.go [hunk 2]
```diff
@@ -23,7 +23,18 @@ import (
...
+	if err == nil {
+		t.Errorf(msg)
+	}
+}
+
+func checkAndReportErr(t *testing.T, msg string, err error) {
```
Comment (Madhu-1) [inline internal/util/topology_test.go:32]: checkAndReportError

File: internal/util/topology_test.go [hunk 3]
```diff
@@ -23,7 +23,18 @@ import (
...
+	if err != nil {
+		t.Errorf(msg+" (%v)", err)
```
Comment (Madhu-1) [inline internal/util/topology_test.go:34]: t.Errorf("%s (%v)", msg,err)

Comment (Yuggupta27) [reply internal/util/topology_test.go:34]: Thanks. I'll make the changes.

**Comment by Yuggupta27** [comment]
@Madhu-1, do we need to retrigger the CI?

**Pull request merged**